frontend/catalyst/compiler.py

```diff
@@ -268,6 +269,63 @@ def get_output_filename(infile):
         return infile.replace(".ll", ".o")
 
 
+class WrapperToCatchExceptions:
+    """Create a wrapper around the generated function to catch runtime exceptions."""
+
+    def __init__(self, params, name):
+        self.params = params
+        self.name = name
+
+    _include = sysconfig.get_paths()["include"]
+    _executable = "clang++"
```

Comment:
Why is this strong preference for the C++ compiler?

Reply:
Just like `c99` there's `cc` which can be used instead of `clang++`. But yes, I am now switching to the alternative implementation that I had in mind which is using libffi to avoid having to compile like this. The reason why I wanted to avoid libffi in the first place is because it introduces a build time dependency. (We need libffi built into wrapper as opposed to just linking).